### ISSUE TITLE
Indent not working correctly when there is a function call inside an if statement

### ISSUE THREAD

**Issue by stoarca** [eslint/eslint#9779]
**Tell us about your environment**

* **ESLint Version:** 4.13.1
* **Node Version:** 6.11.2
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  'env': {
    'browser': true,
    'commonjs': true,
    'es6': true,
    'node': true,
  },
  'extends': 'eslint:recommended',
  'parserOptions': {
    'sourceType': 'module',
    'ecmaVersion': 2017,
  },
 'rules': {
     'indent': ['error', 2], 
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var variableWithAVeryLongNameNumber1 = 1;
var variableWithAnEvenLongerNameNumber2 = 2;
var myFunc = function(a, b) {
  return a + b;
};
if (myFunc(variableWithAVeryLongNameNumber1,
        variableWithAnEvenLongerNameNumber2)) {
  alert('hello world ' + variableWithAVeryLongNameNumber1 + ' ' + variableWithAnEvenLongerNameNumber2);
}
```

```bash
eslint blah.js
```

**What did you expect to happen?**
No warnings or errors. Argument is correctly indented relative to start of `myFunc`.

**What actually happened? Please include the actual, raw output from ESLint.**
blah.js
  7:1  error  Expected indentation of 2 spaces but found 8  indent

It looks like it wants me to indent relative to the start of the line instead of the start of `myFunc`, which is correct in most cases, but not in the case of `for`, `if`, `while`, etc. None of the options available seem to enable this behavior either. Note that if I follow the `2 spaces` instruction that eslint gives me, it would place the second argument at the same indentation as the inside of the if-block, making it look like it's inside the block instead of in the condition.

**Comment by platinumazure** [comment]
I'm not sure how we could reasonably enhance this without introducing some more complexity to the rule (it's already quite complex). That said, I think you could use the [`ignoredNodes` option](https://eslint.org/docs/rules/indent#ignorednodes) to ignore CallExpression nodes within the test/condition of those statements. This would ignore the indentation entirely in those areas.

Unfortunately, I'm not very good with esquery, so I can't give you everything you would need. But you could start with `IfStatement > CallExpression` as an example-- this would ignore function calls that are immediate children of if statements. (I was also playing with `IfStatement > CallExpression.test`, which I had hoped would restrict the selector to only look at function calls inside the if statement's condition, but that didn't bear out how I expected when I was playing around in esquery's demo.)

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
